src/contracts/middleware/BLSSignatureChecker.sol

```diff
@@ -115,26 +116,26 @@ abstract contract BLSSignatureChecker {
 
                 for (uint i = 0; i < nonSignerStakesAndSignature.nonSignerPubkeys.length; i++) {
                     nonSignerPubkeyHashes[i] = nonSignerStakesAndSignature.nonSignerPubkeys[i].hashG1Point();
-                    if (i != 0) {
-                        require(uint256(nonSignerPubkeyHashes[i]) > uint256(nonSignerPubkeyHashes[i - 1]), "BLSSignatureChecker.checkSignatures: nonSignerPubkeys not sorted");
-                    }
+                    // if (i != 0) {
+                    //     require(uint256(nonSignerPubkeyHashes[i]) > uint256(nonSignerPubkeyHashes[i - 1]), "BLSSignatureChecker.checkSignatures: nonSignerPubkeys not sorted");
+                    // }
```

Comment:
is this just a temporary change to deal with ordering being hard?
IIRC this check was key for ensuring no duplicates (we should add a comment to this effect as well)

Reply:
I actually think this is fine. if you subtract the same public key twice, then the amount of stake that has confirmed a fact will only be underestimated

Reply:
> I actually think this is fine. if you subtract the same public key twice, then the amount of stake that has confirmed a fact will only be underestimated

hmm OK I guess. I suppose you wouldn't have a valid signature for the resultant public key anyway, right (in the case where you subtract a signature twice)? Just want to be very sure before deleting anything that was intended as some kind of safety/correctness check.

Reply:
correct, agreed.